### PR TITLE
Add support for grepping key=val pairs on structured log

### PR DESCRIPTION
**Pull request by tuommaki**
As we use structured logging and it's already parsed in formatting, it
makes sense to have ability to grep based on key=value pairs of log
elements.

Common example would be grepping on log level e.g. level=error.

### CODE REVIEW

**Comment by tuommaki** [comment]
My aim is to eventually bundle this PR with unit tests fix PR and merge them as one to `master`. I separated these for easier review.

File: format_test.go [hunk 1]
```diff
@@ -34,14 +39,76 @@ func Test_format(t *testing.T) {
...
+			switch {
+			case err == nil && tc.errorMatcher == nil:
+				// correct; carry on
+			case err != nil && tc.errorMatcher == nil:
+				t.Fatalf("error == %#v, want nil", err)
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case !tc.errorMatcher(err):
+				t.Fatalf("error == %#v, want matching", err)
+			}
```
Comment (kopiczko) [inline format_test.go]: I think it should return when `tc.errorMatcher != nil`

File: format.go [hunk 2]
```diff
@@ -36,11 +37,25 @@ func disable_colors(v bool) {
...
-		return string(text)
+		return "", microerror.Mask(err)
+	}
+
+	if len(grep) > 0 {
+		var notFoundElements []string
+		for k, v := range grep {
+			logVal, exists := m[k]
+			if !exists || logVal != v {
+				notFoundElements = append(notFoundElements, k)
```
Comment (kopiczko) [inline format.go]: How about appending `fmt.Sprintf("%s=%q", k, v)`?

File: format_test.go (hunk 1 above)
Comment (tuommaki) [reply format_test.go]: I'm not sure if that makes much difference, but as you suggest it, I'll make it happen. :)

File: format.go (hunk 2 above)
Comment (tuommaki) [reply format.go]: Sure, I'll change this.

File: format_test.go (hunk 1 above)
Comment (kopiczko) [reply format_test.go]: I missed those are `Fatalf` not `Errorf`. Sorry for the confusion.

**Pull request merged**